x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/event/AnalyticsEvent.java

```diff
@@ -191,17 +187,20 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endObject();
 
             builder.mapContents(payloadAsMap());
+
+            if (userAgent != null) {
+                builder.field(USER_AGENT_FIELD.getPreferredName(), userAgent);
+            }
+
+            if (clientAddress != null) {
+                builder.field(CLIENT_ADDRESS_FIELD.getPreferredName(), NetworkAddress.format(clientAddress));
+            }
```

Comment:
This field are meant to be part of the session object.
Here they are pushed at the root of the object.

Reply:
That was my bad, didn't take a proper look at the index mapping 😞